python/test/test_storage_format.py

```diff
@@ -0,0 +1,273 @@
+import filecmp
+import json
+import pathlib
+import tempfile
+from collections.abc import MutableMapping
+
+import click
+import pytest
+
+from katana.example_data import get_rdg_dataset
+from katana.local import Graph
+
+### Tests Functions ###
+
+
+def test_storage_format_unchanged_local():
+    """
+    load up a known good rdg, store a copy
+    compare the storage format of the known good rdg to our stored copy
+    Attempts to catch the following situations:
+    1) the storage_format_version was changed but the rdg test datasets have not been updated
+    2) the on disk storage format was changed, but the storage_format_version was not
+    3) unstable storage format changes that are in use without the unstable storage format flag
+
+    TODO(emcginnis): this test would be best if we had some way to create and use a 'maximal' RDG, aka one with as many
+        optional features present as possible
+    """
+    orig_rdg = get_rdg_dataset("ldbc_003")
+    orig_graph = Graph(orig_rdg)
+    new_rdg = tempfile.mkdtemp()
```

Comment:
you should use a context manager when using a tempdir to be as explicit as possible

Reply:
using the context manager mean the tmp dir goes away completely after this is done, which really makes debugging a pain. 
I will add a call to cleanup the temp dir at the end on a successful run though.